app/src/main/java/com/davidtomas/taskyapp/features/agenda/presentation/AgendaDetail/components/PhotosComposable.kt

```diff
@@ -0,0 +1,140 @@
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.davidtomas.taskyapp.R
+import com.davidtomas.taskyapp.coreUi.TaskyAppTheme
+import com.davidtomas.taskyapp.features.agenda.domain.model.PhotoModel
+
+@Composable
+fun PhotosComposable(
+    photos: List<PhotoModel>,
+    onAddedPhoto: (String) -> Unit,
+) {
+    val context = LocalContext.current
+    var selectedImageUri by remember {
+        mutableStateOf<Uri?>(null)
+    }
+
+    val singlePhotoPickerLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.PickVisualMedia(),
+        onResult = { uri ->
+            selectedImageUri = uri
+            uri?.toString()?.let { imageUrl ->
+                val flag = Intent.FLAG_GRANT_READ_URI_PERMISSION
+                context.contentResolver.takePersistableUriPermission(uri, flag)
+                onAddedPhoto(imageUrl)
+            }
+        }
+    )
+    Column(verticalArrangement = Arrangement.spacedBy(20.dp)) {
+        Text(
+            text = "Photos",
+            style = MaterialTheme.typography.titleMedium.copy(fontWeight = FontWeight.SemiBold)
+        )
+        Row(
+            modifier = Modifier
+                .fillMaxWidth(),
+        ) {
+            val chunkedTextList = photos.chunked(5) // Break the list into chunks of 3
+
+            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
+                // Iterate over the chunked list
+                for (chunk in chunkedTextList) {
+                    Row(
+                        horizontalArrangement = Arrangement.SpaceAround,
+                        modifier = Modifier.fillMaxWidth()
+                    ) {
+                        // Iterate over each chunk and create a Text composable for each item
```

Comment:
Programmers know what a loop does, so these what-comments don't add a lot of value :D 
I personally only use why-comments to explain why something happens which is not immediately obvious